master/internal/api_resourcepool.go

```diff
@@ -31,7 +36,28 @@ func (a *apiServer) OverwriteRPWorkspaceBindings(
 func (a *apiServer) UnbindRPFromWorkspace(
 	ctx context.Context, req *apiv1.UnbindRPFromWorkspaceRequest,
 ) (*apiv1.UnbindRPFromWorkspaceResponse, error) {
-	return nil, nil
+	curUser, _, err := grpcutil.GetUser(ctx)
+	if err != nil {
+		return nil, err
+	}
+	// Check permissions for all workspaces. Return err if any workspace doesn't have permissions.
+	// No partial unbinding.
+	for _, workspaceID := range req.WorkspaceIds {
+		if err = workspaceauth.AuthZProvider.Get().CanUnBindRPWorkspace(ctx, *curUser,
+			workspaceID); err != nil {
+			return nil, authz.SubIfUnauthorized(err,
+				errors.Errorf(
+					`current user %q doesn't have permissions to unbind
+					 resource pool %q from workspace with id %q.`,
```

Comment:
Hmm this might be tricky because we'll want to differentiate between 404 (workspace does not exist) and not enough permissions. Maybe an easy way around that for now is just something vague like: "current user %q doesn't have permissions to modify resource pool bindings"